### PR TITLE
feat : added signup and some code refactoring

### PR DESCRIPTION
**Pull request by krate27**
Added sign up feature and did some code refactoring

### CODE REVIEW

File: src/Components/Auth/SignUp/signup.css
Comment (xii-vi) [inline src/Components/Auth/SignUp/signup.css:8]: Remove Commented Code

File: src/Components/Auth/SignUp/signup.css
Comment (xii-vi) [inline src/Components/Auth/SignUp/signup.css:63]: be consistent in using the CSS property.
either use rem or px

File: src/Components/Auth/SignUp/signup.css
Comment (xii-vi) [inline src/Components/Auth/SignUp/signup.css:76]: here too

File: src/Components/Fixed-Footer/footer.css
Comment (xii-vi) [inline src/Components/Fixed-Footer/footer.css:22]: remove un-used code

File: src/Components/Fixed-Footer/footer.js
Comment (xii-vi) [inline src/Components/Fixed-Footer/footer.js:31]: either use hex code or name in color

**Review by xii-vi — APPROVED** [review]
**left a few comments. GO through them once before merging**

File: src/Components/Auth/SignUp/signup.js
Comment (suryxks) [inline src/Components/Auth/SignUp/signup.js:16]: `for` should not be used in react as `for` is a reserved keyword in js you should use `htmlFor` instead of it

File: src/App.js
Comment (yaswanthmyneni) [inline src/App.js:4]: suggestion: export all components from one file (index.js) in components folder and import components from it

File: src/Components/Auth/SignUp/signup.css
Comment (yaswanthmyneni) [inline src/Components/Auth/SignUp/signup.css:20]: changes: usually we give units more than 15px in rems unless we need exact lengths.

File: src/Components/Auth/Login/login.css
Comment (theVivek-kumar) [inline src/Components/Auth/Login/login.css:27]: use shorthand CSS for better readability

File: src/Components/Auth/Login/login.css
Comment (theVivek-kumar) [inline src/Components/Auth/Login/login.css:112]: use px

**Review by suryxks — APPROVED** [review]
have left  a suggestion make the changes 
good to merge after making the change

File: src/Components/Auth/Login/login.css
Comment (theVivek-kumar) [inline src/Components/Auth/Login/login.css:153]: use rem here

File: src/Components/Fixed-Footer/footer.js
Comment (yaswanthmyneni) [inline src/Components/Fixed-Footer/footer.js:15]: suggestion: avoid inline styling, use class names for same

File: src/Components/Auth/Login/login.css
Comment (theVivek-kumar) [inline src/Components/Auth/Login/login.css:157]: use rem

File: src/Components/Auth/Login/login.js
Comment (theVivek-kumar) [inline src/Components/Auth/Login/login.js:31]: className is not proper and not well define

File: src/Components/Auth/SignUp/signup.css
Comment (theVivek-kumar) [inline src/Components/Auth/SignUp/signup.css:12]: use rem here

File: src/Components/Auth/SignUp/signup.css
Comment (theVivek-kumar) [inline src/Components/Auth/SignUp/signup.css:20]: use rem

File: src/Context/authContext.js
Comment (yaswanthmyneni) [inline src/Context/authContext.js:48]: suggestion: display errors using console.errors

File: src/Components/Auth/SignUp/signup.css
Comment (theVivek-kumar) [inline src/Components/Auth/SignUp/signup.css:48]: bad practice never semantic tag name

**Review by yaswanthmyneni — APPROVED** [review]
You can merge after changes

File: src/Components/Sidebar/sidebar.css
Comment (theVivek-kumar) [inline src/Components/Sidebar/sidebar.css:230]: plz insert your UI preview

**Review by theVivek-kumar — COMMENTED** [review]
use rem in place of px else is good to merge

**Review by cjinc121 — APPROVED** [review]
some suggestion : 
in styling units, >10px use rems
anchor tank should be replaced with Link 
UI preview prefered

File: src/Components/Fixed-Footer/footer.js
Comment (theVivek-kumar) [reply src/Components/Fixed-Footer/footer.js:15]: do not use inline styling

**Pull request merged**
